lib/irq.c

```diff
@@ -171,8 +171,9 @@ irqs_set_data_eventfd(vfu_ctx_t *vfu_ctx, struct vfio_irq_set *irq_set,
 
             vfu_ctx->irqs->efds[i] = -1;
         }
-        if (*d32 >= 0) {
-            vfu_ctx->irqs->efds[i] = *d32;
+        if (data[j] >= 0) {
+            vfu_ctx->irqs->efds[i] = data[j];
```

Comment:
I actually think this is still much cleaner to return the fd from consume_fd() - you don't need to mention the array and index twice - but I suppose it's not a big deal.

Reply:
I missed the point of `consume_fd`, it makes even more sense now, will fix.